### ISSUE TITLE
MacOS Mojave bug causing hang due to Channel poll() method.

### ISSUE THREAD

**Issue by sam-harding** [noxdafox/pebble#82]
Hello, 

From my testing I believe that the unix_poll method in Channel is susceptible to a bug in MacOS 10.14.6 in which the read only endpoint of a non-duplex multiprocessing.Pipe reports as "ready to write". I think this could also be responsible for [this issue](https://github.com/noxdafox/pebble/issues/50). 

This means that every time `poll.poll` is called from [this line](https://github.com/noxdafox/pebble/blob/58d9b4502820bbec95aca648f6af1e50d8965c17/pebble/pool/channel.py#L48). It returns true as the poll method is reporting the `reader` is "ready to write".

Due to this, the poll function will always return True, meaning that it is possible to end up with a program hanging due to the WorkerManager trying to receive data while the Worker attempts to Send data. In this scenario I have found that the ChannelMutex is unable to acquire the `writer_mutex` RLock, which causes a worker to be unable to send its results.

I would be happy to submit a PR shortly to amend this function so that the poll registers with an eventmask for `POLLIN` and `POLLPRI` so that there isn't a potential issue for MacOS 10.14 users. 

Secondly I noticed that the timeout changed from SECONDS to MILLISECONDS with the [change](https://github.com/noxdafox/pebble/commit/61b7dd292283b6234d4586a50a96e8a1b2200a39) from select.select to select.poll. So the 0.1 timeout now means 0.1ms rather than 0.1s.

## Mojave Poll Bug

This script

```
import select
import multiprocessing as mp

r1, w1 = mp.Pipe(duplex=False)
poll = select.poll()
poll.register(r1)
poll.register(w1)
print(poll.poll())
```

Expected result `[(5, 4)]` (Only the w1 channel is writable)
Actual Result `[(4,4), (5,4)]` (Both channels are writable)

This occurs on multiple machine in 10.14.6

**Comment by noxdafox** [comment]
Hello,

apologies for the late reply. Issue #50 was due to a regression introduced at a certain version of `pebble` in which some critical changes were wrongly implemented during the Pool shutdown. It is not related to this issue.

The change from seconds to millis is definitely something I overlooked myself and it should be fixed. We want to test for new data every `0.1` second rather than `0.0001`. Would you mind fixing that issue as well in your PR?

**Comment by sam-harding** [comment]
Yes sure, I will get it done this week.

**Comment by sam-harding** [comment]
Hello, I have updated the PR with a timeout conversion.

**Comment by noxdafox** [comment]
Thanks!

Added a comment in there.

**Comment by noxdafox** [comment]
Issue resolved in version `4.6.2`.

Thanks for your contribution!
